rca/static_src/javascript/programmes/components/ProgrammesResults/ProgrammesResults.js

```diff
@@ -0,0 +1,126 @@
+
+import Icon from '../Icon/Icon';
+import ProgrammeTeaser from './ProgrammeTeaser';
+
+const getResultsCount = (count) => {
+    switch (count) {
+        case 0: {
+            return 'No results match your search';
+        }
+        case 1: {
+            return '1 result matches your search';
+        }
+        default: {
+            return `${count} results match your search`;
+        }
+    }
+};
+
+/**
+ * A list of programmes matching a search or filter.
+ * The list auto-magically appears when matches are found.
+ */
+const ProgrammesResults = ({ programmes, hasActiveSearch }) => {
+    const [activeProgramme, setActiveProgramme] = useState(null);
+    if (!hasActiveSearch) {
+        return null;
+    }
+
+    const count = getResultsCount(programmes.length);
+    return (
+        <>
+            <div className="programmes-results bg bg--dark section">
+                <div className="grid">
+                    <div className="programmes-results__actions">
+                        <button
+                            type="button"
+                            className="button programmes-results__back body body--one"
+                        >
+                            <Icon
+                                name="arrow"
+                                className="programmes-results__back__icon"
+                            />
+                            <span className="programmes-results__back__text">
+                                Back
+                            </span>
+                        </button>
+                    </div>
+                    <p
+                        className="heading heading--five programmes-results__count"
+                        role="alert"
+                    >
+                        {count}
+                    </p>
+                </div>
+                {programmes.length === 0 ? null : (
+                    <div className="grid">
+                        <div className="programmes-results__list">
+                            {programmes.map((prog) => {
+                                return (
+                                    <ProgrammeTeaser
+                                        key={prog.id}
+                                        programme={prog}
+                                        onFocus={() =>
+                                            setActiveProgramme(prog.id)
+                                        }
+                                        onMouseOver={() =>
+                                            setActiveProgramme(prog.id)
+                                        }
+                                    />
+                                );
+                            })}
+                        </div>
+                        <div className="programmes-results__images">
+                            <div className="programmes-results__images-sticky">
+                                {programmes.map((prog, i) => {
```

Comment:
This is probably the most complex bit of this UI. We want the images to be shown in a fixed position for all items, with only the currently "active" image visible (hover, focus, or otherwise first of the results). And this has to be sticky.

We render all of the images by default so they preload when the search results display, and have a transition between them. Then the active image has a class applied to it.

Reply:
Looks slick!